dvc/remote/base.py

```diff
@@ -872,10 +899,18 @@ def cache_exists(self, checksums, jobs=None, name=None):
         # cache_exists() (see ssh, local)
         assert self.TRAVERSE_PREFIX_LEN >= 2
 
-        if len(checksums) == 1 or not self.CAN_TRAVERSE:
-            return self._cache_object_exists(checksums, jobs, name)
+        checksums = set(checksums)
+        indexed_checksums = set(self.index.intersection(checksums))
+        checksums -= indexed_checksums
+        logger.debug(
+            "Matched {} indexed checksums".format(len(indexed_checksums))
```

Comment:
minor: please try to wrap dynamic part of the message in single-quotes, it makes it much easier to visually parse. And to be consistent too :slightly_smiling_face: 
```suggestion
            "Matched '{}' indexed checksums".format(len(indexed_checksums))
```